raw/fetch.py

```diff
@@ -0,0 +1,224 @@
+        first_day_third_prev (datetime.date): The first day of the third previous month.
+        last_day_third_prev (datetime.date): The last day of the third previous month.
+        sl_year (str): The year of the third previous month in 'YYYY' format.
+        sl_month (str): The month of the third previous month in 'MM' format.
+        sl_first_date (str): The first date of the third previous month in 'DD' format.
+        sl_last_date (str): The last date of the third previous month in 'DD' format.
+    """
+
+    with open(config_file, "r") as file:
+        lines = file.readlines()
+
+    # Update the specified field with the new value
+    updated_lines = []
+    selection_line_found = False
+    for line in lines:
+        if not selection_line_found and line.strip() == "[selection]":
+            updated_lines.append(f"{additional_content}\n")
+            selection_line_found = True
+
+        if single_level_file:
+            if line.startswith("year"):
+                line = f"year={sl_year}\n"
+            if line.startswith("month"):
+                line = f"month={sl_month}\n"
+            if line.startswith("day"):
+                line = f"day={sl_first_date}/to/{sl_last_date}\n"
+        elif line.startswith(field_name):
+            if co_file:
+                line = (
+                    f"{field_name}={first_day_first_prev}/to/{last_day_first_prev}\n"
+                )
+            else:
+                line = (
+                    f"{field_name}={first_day_third_prev}/to/{last_day_third_prev}\n"
+                )
+        updated_lines.append(line)
+
+    with open(config_file, "w") as file:
+        file.writelines(updated_lines)
+
+
+def get_month_range(date: datetime.date) -> Tuple[datetime.date, datetime.date]:
+    """Return the first and last date of the month from the input date.
+
+    Parameters:
+        date (datetime.date): The input date.
+
+    Returns:
+        tuple: A tuple containing the first and last date of the month as 
+                datetime.date objects.
+    """
+    last_day = date.replace(day=1) - datetime.timedelta(days=1)
+    first_day = last_day.replace(day=1)
+    return first_day, last_day
+
+
+def get_single_level_dates(first_day: datetime.date, 
+                           last_day: datetime.date) -> Tuple[str, str, str, str]:
+    """Return the year, month, first date, and last date of the input month.
+
+    Parameters:
+        first_day (datetime.date): The first day of the month.
+        last_day (datetime.date): The last day of the month.
+
+    Returns:
+        tuple: A tuple containing the year, month, first date, and last date of the 
+                month as strings.
+    """
+    year, month = str(first_day)[:4], str(first_day)[5:7]
+    first_date, last_date = str(first_day)[8:], str(last_day)[8:]
+    return (year, month, first_date, last_date)
+
+
+def get_previous_month_dates() -> (Tuple[datetime.date, datetime.date, datetime.date,
```

Comment:
I recommend returning a single dictionary, given how long this tuple is.

Reply:
Code updated as per your suggestion. Thanks for the input!